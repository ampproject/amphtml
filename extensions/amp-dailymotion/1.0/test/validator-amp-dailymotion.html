<!DOCTYPE html>
<html âš¡>
  <!-- prettier-ignore -->
  <head>
    <meta charset="utf-8">
    <title>amp-dailymotion example</title>
    <link rel="canonical" href="/examples/amp-dailymotion.html">
    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
    <style amp-custom>
      amp-dailymotion {
        color: red;
      }
    </style>
    <script async custom-element="amp-dailymotion" src="https://cdn.ampproject.org/v0/amp-dailymotion-1.0.js"></script>
    <script async src="https://cdn.ampproject.org/v0.js"></script>
  </head>
  <body>
    <!-- Valid. -->
    <amp-dailymotion
      data-videoid="x3rdtfy"
      width="500"
      height="281">
    </amp-dailymotion>

    <!-- Valid: with custom url attributes. -->
    <amp-dailymotion
      data-videoid="x3rdtfy"
      data-endscreen-enable="false"
      data-sharing-enable="false"
      data-ui-highlight="444444"
      data-ui-logo="false"
      data-info="false"
      id="myVideo"
      width="640"
      height="360">
    </amp-dailymotion>

    <!-- Valid: with autoplay. -->
    <amp-dailymotion
      autoplay
      data-videoid="x3rdtfy"
      width="500"
      height="281"
      layout="responsive">
    </amp-dailymotion>

    <!-- Invalid: missing data-videoid. -->
    <amp-dailymotion
      width="500"
      height="281">
    </amp-dailymotion>

    <!-- Invalid: missing dimensions. -->
    <amp-dailymotion data-videoid="x3rdtfy"></amp-dailymotion>

    <!-- Invalid: data-videoid must be a video id not a URL. -->
    <amp-dailymotion
      data-videoid="https://www.dailymotion.com/embed/video/x3rdtfy?api=1&html=1&app=amp"
      width="500"
      height="281">
    </amp-dailymotion>
  </body>
</html>
