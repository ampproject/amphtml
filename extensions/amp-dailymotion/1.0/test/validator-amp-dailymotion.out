FAIL
|  <!DOCTYPE html>
|  <html âš¡>
|    <!-- prettier-ignore -->
|    <head>
|      <meta charset="utf-8">
|      <title>amp-dailymotion example</title>
|      <link rel="canonical" href="/examples/amp-dailymotion.html">
|      <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|      <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|      <style amp-custom>
|        amp-dailymotion {
|          color: red;
|        }
|      </style>
|      <script async custom-element="amp-dailymotion" src="https://cdn.ampproject.org/v0/amp-dailymotion-1.0.js"></script>
|      <script async src="https://cdn.ampproject.org/v0.js"></script>
|    </head>
|    <body>
|      <!-- Valid. -->
|      <amp-dailymotion
|        data-videoid="x3rdtfy"
|        width="500"
|        height="281">
|      </amp-dailymotion>
|
|      <!-- Valid: with custom url attributes. -->
|      <amp-dailymotion
|        data-videoid="x3rdtfy"
|        data-endscreen-enable="false"
|        data-sharing-enable="false"
|        data-ui-highlight="444444"
|        data-ui-logo="false"
|        data-info="false"
|        id="myVideo"
|        width="640"
|        height="360">
|      </amp-dailymotion>
|
|      <!-- Valid: with autoplay. -->
|      <amp-dailymotion
|        autoplay
|        data-videoid="x3rdtfy"
|        width="500"
|        height="281"
|        layout="responsive">
|      </amp-dailymotion>
|
|      <!-- Invalid: missing data-videoid. -->
|      <amp-dailymotion
>>     ^~~~~~~~~
amp-dailymotion/1.0/test/validator-amp-dailymotion.html:49:4 The mandatory attribute 'data-videoid' is missing in tag 'amp-dailymotion'. (see https://amp.dev/documentation/components/amp-dailymotion/)
|        width="500"
|        height="281">
|      </amp-dailymotion>
|
|      <!-- Invalid: missing dimensions. -->
|      <amp-dailymotion data-videoid="x3rdtfy"></amp-dailymotion>
>>     ^~~~~~~~~
amp-dailymotion/1.0/test/validator-amp-dailymotion.html:55:4 Incomplete layout attributes specified for tag 'amp-dailymotion'. For example, provide attributes 'width' and 'height'. (see https://amp.dev/documentation/components/amp-dailymotion/)
|
|      <!-- Invalid: data-videoid must be a video id not a URL. -->
|      <amp-dailymotion
>>     ^~~~~~~~~
amp-dailymotion/1.0/test/validator-amp-dailymotion.html:58:4 The attribute 'data-videoid' in tag 'amp-dailymotion' is set to the invalid value 'https://www.dailymotion.com/embed/video/x3rdtfy?api=1&html=1&app=amp'. (see https://amp.dev/documentation/components/amp-dailymotion/)
|        data-videoid="https://www.dailymotion.com/embed/video/x3rdtfy?api=1&html=1&app=amp"
|        width="500"
|        height="281">
|      </amp-dailymotion>
|    </body>
|  </html>